### PR TITLE
Add public libcudf match_dictionaries API

### PR DESCRIPTION
**Pull request by davidwendt**
This PR creates a public API for the internal libcudf `cudf::dictionary::detail::match_dictionaries` function to help with transitioning the cudf python CategoricalColumn over to using the libcudf dictionary column.

No function has changed or been added but this PR does add a formal gtest for the new public API.

### CODE REVIEW

**Review by ttnghia — APPROVED** [review]

File: cpp/include/cudf/dictionary/update_keys.hpp [hunk 1]
```diff
@@ -139,6 +139,20 @@ std::unique_ptr<column> set_keys(
...
 
+/**
+ * @brief Create new dictionaries that have keys merged from the input dictionaries.
+ *
+ * This will concatenate the keys for each dictionary and then call `set_keys` on each.
+ * The result is a vector of new dictionaries with a common set of keys.
+ *
+ * @param input Dictionary columns to match keys.
+ * @param mr Device memory resource used to allocate the returned column's device memory.
+ * @return New dictionary column.
```
Comment (vuule) [inline cpp/include/cudf/dictionary/update_keys.hpp:150]:
```suggestion
 * @return New dictionary columns.
```

File: cpp/include/cudf/dictionary/update_keys.hpp [hunk 2]
```diff
@@ -139,6 +139,20 @@ std::unique_ptr<column> set_keys(
...
+ */
+std::vector<std::unique_ptr<column>> match_dictionaries(
+  std::vector<dictionary_column_view> input,
```
Comment (vuule) [inline cpp/include/cudf/dictionary/update_keys.hpp:153]:
```suggestion
  host_span<dictionary_column_view const> input,
```

**Review by vuule — CHANGES_REQUESTED** [review]
a few very small suggestions

**Review by vuule — APPROVED** [review]
Thank you :)

**Review by devavret — APPROVED** [review]

**Comment by galipremsagar** [comment]
rerun tests

**Comment by davidwendt** [comment]
@gpucibot merge

**Pull request merged**